packages/engine-server/src/config.ts

```diff
@@ -247,7 +247,10 @@ export class DConfig {
   static readConfigSync(wsRoot: string) {
```

Comment:
can we cache this in a global object instead of reading it from the fs everytime?

Reply:
We currently don't have a file watcher for config files. This is why if a user manually updates the config, it doesn't get reflected in the engine.config prop.

Once we add watcher support, we can cache, but I'm not sure if thats necessary right now

Reply:
I don't think the performance gain is worth the complication right now of a file watcher - fs reading on a single file isn't so bad, we do sync reads into FS all over the place anyway. 

If anything, one optimization we can do is to have an async version `readConfig` where consumers can use the async version if they want to / can.

Reply:
I also feel reading from `fs` is not bad. To compromise, I'll add a global config, but the default behavior is to read from fs. The expensive calls like `getDecorations` will use the cache while the others read from fs.

The global config also doesn't work well with the tests, hence why I'm making default read from fs